### PR TITLE
Update `GrafanaDashboardAggregator`

### PR DESCRIPTION
**Pull request by rbarry82**
## Issue
Some dashboards were not passed appropriately through `cos-proxy`, since some "Juju topology" was added (with different variable names, such as `$app` instead of `$juju_application`). Add more mangling so they behave.

### CODE REVIEW

File: lib/charms/grafana_k8s/v0/grafana_dashboard.py [hunk 1]
```diff
@@ -1716,11 +1733,17 @@ def _handle_reactive_dashboards(self, event: RelationEvent) -> Optional[Dict]:
...
+            )
+
+            # Yank out "new"+old LMA topology
+            dash = re.sub(r',?juju_application=~"$app"', "", dash)
 
             from jinja2 import Template
 
             content = _encode_dashboard_content(
-                Template(dash).render(host=event.unit.name, datasource="prometheus")
+                Template(dash).render(host=r"$host", datasource="${prometheusds}")
```
Comment (simskij) [inline lib/charms/grafana_k8s/v0/grafana_dashboard.py:1746]: Should the datasource also be a raw string?

**Review by simskij — APPROVED** [review]

**Review by sed-i — COMMENTED** [review]
:shrug: 

Test?

File: lib/charms/grafana_k8s/v0/grafana_dashboard.py (hunk 1 above)
Comment (dstathis) [reply lib/charms/grafana_k8s/v0/grafana_dashboard.py:1746]: Why is it raw at all? Does "$" do something I am unaware of?

File: lib/charms/grafana_k8s/v0/grafana_dashboard.py [hunk 2]
```diff
@@ -1751,10 +1774,10 @@ def _maybe_get_builtin_dashboards(self, event: RelationEvent) -> Dict:
 
         if dashboards_path:
 
-            def _is_dashboard(p: Path) -> bool:
-                return p.is_file() and p.name.endswith((".json", ".json.tmpl", ".tmpl"))
+            def _is_dashbaord(p: Path) -> bool:
+                return p.is_file and p.name.endswith((".json", ".json.tmpl", ".tmpl"))
```
Comment (dstathis) [inline lib/charms/grafana_k8s/v0/grafana_dashboard.py:1778]: typo here?

File: lib/charms/grafana_k8s/v0/grafana_dashboard.py [hunk 3]
```diff
@@ -1751,10 +1774,10 @@ def _maybe_get_builtin_dashboards(self, event: RelationEvent) -> Dict:
...
 
-            for path in filter(_is_dashboard, Path(dashboards_path).glob("*")):
+            for path in filter(_is_dashbaord, Path(dashboards_path).glob("*")):
```
Comment (dstathis) [inline lib/charms/grafana_k8s/v0/grafana_dashboard.py:1780]: Also type here?

**Review by dstathis — APPROVED** [review]

**Pull request merged**